src/vs/platform/clipboard/browser/clipboardService.ts

```diff
@@ -28,31 +34,64 @@ export class BrowserClipboardService implements IClipboardService {
 		try {
 			return await navigator.clipboard.writeText(text);
 		} catch (error) {
-			console.error(error);
-		}
-
-		// Fallback to textarea and execCommand solution
-
-		const activeElement = document.activeElement;
-
-		const textArea: HTMLTextAreaElement = document.body.appendChild($('textarea', { 'aria-hidden': true }));
-		textArea.style.height = '1px';
-		textArea.style.width = '1px';
-		textArea.style.position = 'absolute';
+			if (!isSafari) {
+				console.error(error);
+			} else {
+				const showResult = await this.dialogService.show(
+					Severity.Warning,
+					localize('unableToCopy', "The browser interrupted the copying of text to the clipboard. Press 'Copy' to copy it anyway."),
+					[
```

Comment:
Looking at the fallback below I wonder if we actually need the warning dialog here? Extensions can initiate copy-paste whenever they want, esp without explicit user interaction, and I would be surprised if a modal interrupts my flow.

Reply:
@jrieken the dialog is needed to get user interaction so that the browser allows the operation

Reply:
Yeah, I know about the limitation about user-gesture with `clipboard.writeText` but I wonder about the textarea trick? That's outside the gesture, right? Will that work just like that?

Reply:
It doesn't seem to work in Safari from my testing.

Reply:
Hm, I understand. Thanks for clarifying... My concern is about extensions that very aggressively invoke `writeText` - that will cause an avalanche of dialogs and we should probably consider something like "don't ask again"

Reply:
Yeah I agree. A big modal in the face is not great. Interestingly @bpasero already partially fixed this behavior here (for readText not writeText):
https://github.com/microsoft/vscode/blob/de32c181d11b699bfeafaef36ec56eabab12e101/src/vs/workbench/services/clipboard/browser/clipboardService.ts#L53-L69

He used a notification rather than a dialog. I wonder if that is better to use in the clipboard case since we have made efforts to prevent avalanches of notifications from happening?